api/src/main/java/io/strimzi/api/kafka/model/balancing/KafkaRebalanceState.java

```diff
@@ -53,23 +55,37 @@
      *     <dt>ProposalReady</dt><dd>If the user sets annotation strimzi.io/rebalance=refresh and the proposal is already ready.</dd>
      * </dl>
      */
-    Stopped,
+    Stopped(List.of(KafkaRebalanceAnnotation.refresh)),
     /**
      * There's been some error.
      * Transitions to:
      * <dl>
      *     <dt>New</dt><dd>If the error was caused by the resource itself that was fixed by the user.</dd>
      * </dl>
      */
-    NotReady,
+    NotReady(List.of(KafkaRebalanceAnnotation.refresh)),
     /**
      * The rebalance is complete and there is no transition from this state.
      * The resource is eligible for garbage collection after a configurable delay.
      * There is no transition from this state to a new one.
      */
-    Ready,
+    Ready(List.of(KafkaRebalanceAnnotation.refresh, KafkaRebalanceAnnotation.none)),
```

Comment:
Again, this enum is the source of truth for what annotations are valid. `none` is not a valid annotation.